dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,265 @@
+import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.links.model.Link;
+import com.dotmarketing.portlets.workflows.business.WorkflowAPI;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
+import com.dotmarketing.portlets.workflows.model.WorkflowScheme;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilHTML;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import io.vavr.control.Try;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.dotmarketing.business.PermissionAPI.PERMISSION_READ;
+
+/**
+ * Encapsulates the logic to interact with the Browser App
+ */
+public interface BrowserAPI {
+
+    /**
+     * @deprecated see {@link #getFolderContent(BrowserQuery)}
+     * @param usr
+     * @param folderId
+     * @param offset
+     * @param maxResults
+     * @param filter
+     * @param mimeTypes
+     * @param extensions
+     * @param showArchived
+     * @param noFolders
+     * @param onlyFiles
+     * @param sortBy
+     * @param sortByDesc
+     * @param excludeLinks
+     * @param languageId
+     * @return
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Deprecated
+    public default Map<String, Object> getFolderContent (final User usr, final String folderId, final int offset, final int maxResults, final String filter, final List<String> mimeTypes,
+                                                 final List<String> extensions, final boolean showArchived, final boolean noFolders, final boolean onlyFiles, final String sortBy,
+                                                 final boolean sortByDesc, final boolean excludeLinks, final long languageId) throws DotSecurityException, DotDataException {
+    	return getFolderContent(usr, folderId, offset, maxResults, filter, mimeTypes, extensions, true, showArchived, noFolders, onlyFiles, sortBy, sortByDesc, excludeLinks, languageId);
+    }
+
+    /**
+     * @deprecated see {@link #getFolderContent(BrowserQuery)}
+     * @param usr
+     * @param folderId
+     * @param offset
+     * @param maxResults
+     * @param filter
+     * @param mimeTypes
+     * @param extensions
+     * @param showArchived
+     * @param noFolders
+     * @param onlyFiles
+     * @param sortBy
+     * @param sortByDesc
+     * @param languageId
+     * @return
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Deprecated
+    public default Map<String, Object> getFolderContent (final  User usr, final String folderId, final int offset, final int maxResults, final String filter, final List<String> mimeTypes,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary modifier 'public' on method 'getFolderContent': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)